Content/Tools/Developer/APITool.cs

```diff
@@ -0,0 +1,50 @@
+﻿using DragonLens.Core.Systems.ThemeSystem;
+using DragonLens.Core.Systems.ToolSystem;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework.Input;
+using ReLogic.Content;
+using System;
+using Terraria;
+using Terraria.Localization;
+using Terraria.ModLoader;
+
+internal sealed class APITool : Tool
+{
+	private readonly Asset<Texture2D> _icon;
+	private readonly Action _click;
+	private readonly string _key;
+
+	public APITool(Mod parent, string key, Asset<Texture2D> icon, Action onClick)
+	{
+		_key = key;
```

Comment:
Key likely needs the calling mod's internal name appended to it somewhere to ensure no collisions